### PR TITLE
Feat/handle translation

### PR DESCRIPTION
**Pull request by mathieurobert1**
Handle the translation in the config file for Primitives(Sphere)/camera/Lights.

image

Also adding an config header RaytracerRules to handle all the important defines of the project.

### CODE REVIEW

File: include/RaytracerRules.hpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+** EPITECH PROJECT, 2024
+** Raytracer
+** File description:
+** Rules header
+*/
+
+// This file handle all the important/common rules
+// of the project
```
Comment (Njord201) [inline include/RaytracerRules.hpp:9]: Move the description of the file im the header of the file.

File: src/Core/Parser/Scene.cpp [hunk 2]
```diff
@@ -145,6 +163,15 @@ int Raytracer::Scene::_parseLightsSetting(const libconfig::Setting &lights)
 
             newPoint->setPosition(position);
             newPoint->setDiffuseMultiplier(diffuse);
+
+            if (lightArrayPoint[index].exists("translation")) {
+                libconfig::Setting& translation = lightArrayPoint[index].lookup("translation");
+                Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+                Math::Vector3D neworigin = newPoint->getPosition();
+                neworigin.translate(trans);
+                newPoint->setPosition(neworigin);
```
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:172]: Rename the variable to newOrigin.

```suggestion
                Math::Vector3D newOrigin = newPoint->getPosition();
                newOrigin.translate(trans);
                newPoint->setPosition(newOrigin);
```

File: src/Core/Parser/Scene.cpp [hunk 3]
```diff
@@ -76,6 +76,15 @@ int Raytracer::Scene::_parseCameraSetting(const libconfig::Setting &camera)
         Math::Vector3D rotation(_parseValue(rotationX), _parseValue(rotationY), _parseValue(rotationZ));
         this->_camera.setRotation(rotation);
     }
+
+    if (camera.exists("translation")) {
+        libconfig::Setting& translation = camera.lookup("translation");
+        Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+        Math::Vector3D neworigin = this->_camera.getOrigin();
+        neworigin.translate(trans);
+        this->_camera.setOrigin(neworigin);
```
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:85]: Rename the variable to newOrigin.

```suggestion
        Math::Vector3D newOrigin = this->_camera.getOrigin();
        newOrigin.translate(trans);
        this->_camera.setOrigin(newOrigin);
```

File: src/Core/Parser/Scene.cpp [hunk 4]
```diff
@@ -111,6 +120,15 @@ int Raytracer::Scene::_parsePrimitiveSetting(const libconfig::Setting &primitive
                 std::shared_ptr<FlatColor> materialPtr = std::make_shared<FlatColor>(color["r"], color["g"], color["b"]);
                 newSphere->setMaterial(materialPtr);
             }
+
+            if (sphereArray[index].exists("translation")) {
+                libconfig::Setting& translation = sphereArray[index].lookup("translation");
+                Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+                Math::Vector3D neworigin = newSphere->getOrigin();
+                neworigin.translate(trans);
+                newSphere->setOrigin(neworigin);
```
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:129]: Rename the variable to newOrigin.

```suggestion
                Math::Vector3D newOrigin = newSphere->getOrigin();
                newOrigin.translate(trans);
                newSphere->setOrigin(newOrigin);
```

File: src/Core/Parser/Scene.cpp (hunk 3 above)
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:85]: Rename the varaible to newOrigin.
```suggestion
        Math::Vector3D newOrigin = this->_camera.getOrigin();
        newOrigin.translate(trans);
        this->_camera.setOrigin(newOrigin);
```

File: src/Core/Parser/Scene.cpp [hunk 5]
```diff
@@ -169,6 +196,15 @@ int Raytracer::Scene::_parseLightsSetting(const libconfig::Setting &lights)
             newDirectional->setPosition(position);
             newDirectional->setDirection(direction);
             newDirectional->setDiffuseMultiplier(diffuse);
+
+            if (lightArrayDirectional[index].exists("translation")) {
+                libconfig::Setting& translation = lightArrayDirectional[index].lookup("translation");
+                Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+                Math::Vector3D neworigin = newDirectional->getPosition();
+                neworigin.translate(trans);
+                newDirectional->setPosition(neworigin);
```
Comment (Njord201) [inline src/Core/Parser/Scene.cpp:205]: Rename the variable to newOrigin.

```suggestion
                Math::Vector3D newOrigin = newDirectional->getPosition();
                newOrigin.translate(trans);
                newDirectional->setPosition(newOrigin);
```

File: include/RaytracerRules.hpp [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+#pragma once
+
+#define VOID_COLOR Math::Vector3D(0,0,0)
```
Comment (Njord201) [inline include/RaytracerRules.hpp:13]: I'm not agree with the name of VOID_COLOR, maybe we can translate it by BLACK_COLOR ?

**Review by Njord201 — COMMENTED** [review]
Some request, and one discussion about the name of VOID_COLOR

**Review by Njord201 — CHANGES_REQUESTED** [review]
Some request, and one discussion about the name of VOID_COLOR

File: include/RaytracerRules.hpp (hunk 6 above)
Comment (mathieurobert1) [reply include/RaytracerRules.hpp:13]: No it's the color of the void, black is the color of the void but black is not the void. You understand ?

Comment (Njord201) [reply include/RaytracerRules.hpp:13]: I understand

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**